### PR TITLE
added group_by property

### LINKED ISSUE #169: Scenario Selector - Group scenarios by year/month/week/day
## Feature Improvement

With many scenarios on a Daily frequency, the Scenario Selector will be very long and hard to navigate. A way to solve this issue would be to group the cycles by weeks, months, and years if needed.

In a nutshell, this is a way to group scenarios in the scenario selector when they are too many.

## Motivation
The scenario selector becomes too large and hard to navigate with many scenarios or cycles. This feature helps end-users navigate within the list of scenarios.

## Expected behavior
A new group_by property should be exposed to activate the feature. 
The property could accept an auto value to let Taipy decide the categories to group the scenarios. Otherwise, a list of possible groups among day, week, month, quarter, and year could be passed.
Let’s imagine we have daily scenarios, and the groups provided are year, month, and week. The groups should be organized as follows: 

```
2024
    |___April
    |       |___week 18
    |       |       |___Cycle - 29 April 2024
    |       |                |___ Scenario A
    |       |                |___ Scenario B
    |       |                |___ Scenario C
    |       |
    |       |___week 17
    |       |___week 16
    |       |___week 15
    |       |___week 14
    |___March
    |___February
    |___January
           |___week 4
           |___week 3
           |___week 2
           |___week 1
                   |___Cycle - 07/01/2024
                   |___Cycle - 06/01/2024
                   |___Cycle - 05/01/2024
                   |___Cycle - 04/01/2024
                   |        |___ Scenario FOO
                   |        |___ Scenario BAR
                   |        |___ Scenario BAZ
                   |
                   |___Cycle - 03/01/2024
                   |___Cycle - 02/01/2024
                   |___Cycle - 01/01/2024
2023
    |___December
    |        |___week 52
    |___November

```

Without Cycles, Grouped by Month (and “Year” appears as well for months to be unique)
```
2024
    |___April
    |       |___ Scenario A
    |       |___ Scenario B
    |       |___ Scenario C
    |___March
    |___February
    |___January
2023
    |___December
    |       |___ Scenario X
    |       |___ Scenario Y
    |___November

```
With auto-on and multiple scenarios per day
```
2024
    |___April
    |       |___Day 29
    |       |       |___ Scenario A
    |       |       |___ Scenario B
    |       |       |___ Scenario C
    |       |___Day 28
    |       |___Day 27
    |       |___Day 26
    |___March
    |___February
    |___January
2023
    |___December
    |        |___Day 31
    |        |___Day 30

```

## Usage

```
<| {scenario} | scenario_selector | group_by=auto |>
<| {scenario} | scenario_selector | group_by=cycle |>
<| {scenario} | scenario_selector | group_by=date |>
```

### PR DESCRIPTION
**Pull request by eleensmathew**
I made some changes to add group_by property. Please check and let me know where I'm going wrong? All kinds of pointers are appreciated. Fixes #169

### CODE REVIEW

**Comment by jrobinAV** [comment]
Can you create unit tests of the new method `get_grouped_scenario_list(entities, group_by)`?
Unit tests help to:

- make the PR easy to review
- improve the readability and maintainability of the code
- document how a method should be used
- avoid future bugs or regressions

Thank you.

File: taipy/gui_core/_context.py [hunk 1]
```diff
@@ -352,11 +352,61 @@ def get_filtered_scenario_list(
...
+
+        return grouped_scenarios
+            
+
     def get_scenarios(
         self,
         scenarios: t.Optional[t.List[t.Union[Cycle, Scenario]]],
         filters: t.Optional[t.List[t.Dict[str, t.Any]]],
         sorts: t.Optional[t.List[t.Dict[str, t.Any]]],
+        group_by: t.Optional[str],
```
Comment (FredLL-Avaiga) [inline taipy/gui_core/_context.py:415]: How do you specify that parameter ?
I think you need to add it as an external property in _GuiCoreLib.py

**Review by FredLL-Avaiga — CHANGES_REQUESTED** [review]
not finished yet :-)

File: taipy/gui_core/_context.py (hunk 1 above)
Comment (eleensmathew) [reply taipy/gui_core/_context.py:415]: Ok thanks for pointing that out. Will make the changes.

**Comment by FredLL-Avaiga** [comment]
@eleensmathew Are you still working on this ?
Can you address the remaining questions and failed checks ?
If you're not, can you close this PR ?

**Comment by eleensmathew** [comment]
Feel free to assign it to someone else

**Comment by FredLL-Avaiga** [comment]
> Feel free to assign it to someone else

We can unassign the issue no problem, but we don't want hanging PRs. So please close it if you're not planning on working on the subject anymore.

**Pull request closed without merging**